framework/selftest.cpp

```diff
@@ -926,6 +926,25 @@ static const kvm_config_t *selftest_kvm_config_real_16bit_fail()
 }
 #endif // __linux__ && x86-64
 
+static int selftest_inject_idle(struct test *test, int cpu)
+{
+    auto loop_start = std::chrono::steady_clock::now();
+    TEST_LOOP(test, 1) {
+        for (int i = 0; i < 10000; ++i) {
+            int a_variable = cpu * 2;
+            a_variable += 1;
```

Comment:
The compiler will optimise this out of existence, along the entire loop.

Append:
```c++
    // use the result so the compiler doesn't optimize this away
    __asm__ ("" : "+gv" (a_variable));
```